napari/utils/io.py

```diff
@@ -232,25 +241,26 @@ def magic_imread(filenames, *, use_dask=None, stack=True):
     images = []
     shape = None
     for filename in filenames_expanded:
-        if guess_zarr_path(filename):
-            image, zarr_shape = read_zarr_dataset(filename)
-            # 1D images are currently unsupported, so skip them.
-            if len(zarr_shape) == 1:
-                continue
-            if shape is None:
-                shape = zarr_shape
-        else:
-            if shape is None:
-                image = imread(filename)
-                shape = image.shape
-                dtype = image.dtype
-            if use_dask:
-                image = da.from_delayed(
-                    delayed(imread)(filename), shape=shape, dtype=dtype
-                )
-            elif len(images) > 0:  # not read by shape clause
-                image = imread(filename)
-        images.append(image)
+        with file_or_url_context(filename) as filename:
```

Comment:
actually, since only tifffile needs this, maybe let's be a bit more cautious and only use this context in `utils.io.imread` when the extension is a tiff.  That avoids unnecessary network activity for stuff like imageio and zarr (which I haven't looked closely at to determine url compatibility)